Script/Selenops.swift

```diff
@@ -81,5 +81,59 @@ func parse(document: String, url: URL) {
   collectLinks().forEach { pagesToVisit.insert($0) }
 }
 
+func validateUrl(urlString: String) -> Bool {
+  // create NSURL instance
+  if NSURL(string: urlString) != nil {
+    // check if your application can open the NSURL instance
+    return true
+  }
+  return false
+}
+
+//error out from bad parameters
+func errorOut(message: String) {
+  print("🚫 Error: \(message)");
+  exit(1);
+}
+
+
+/**
+ * April 13, 2017, added support for dynamic URL, search word, and max page visits.
+ * Validate and set URL below as well as set the search word and max page visits.added
+ * Max page visits is optional, it defaults to 10. If no arguments are added, then
+ * the pre-defined arguments are used.
+ */
+let args:[String] = CommandLine.arguments
+//lets check if they just use type -help or help
+if (args.count == 2) {
+  if args[1] == "help" || args[1] == "-help" {
+    print("usage: swift selenops [startUrl searchWord [maxNumberOfPagesToVisit]]");
+    print("\t-Either no arguments can be used, two arguments can be used (startUrl");
+    print("\t and searchWord), or all three arguments can be used.");
+    exit(0);
+  }
+}
+//we should have a URL and a word
+if (args.count >= 3) {     //validate and set the word
+  if validateUrl(urlString: args[1]) {
```

Comment:
Instead of creating two ``func`` (``errorOut`` and ``validateUrl``), why not use ``guard let``?

    guard let url = URL(string: args[1]) else { 
      print("bad url")
      return
     } 
    startUrl = url

Even better, you can let ``startUrl`` be a constant (``let`` instead of a ``var``) and assign its value only after checking whether there are arguments or not :)